vaadin-testbench-shared/src/main/java/com/vaadin/testbench/ElementQuery.java

```diff
@@ -233,6 +412,42 @@ protected SearchContext getContext() {
         return searchContext;
     }
 
+    /**
+     * Executes the search and returns an element having the given unique id.
+     * <p>
+     *     This selector expects the id to be unique.
+     *     If there are duplicate ids, this selector will
+     *     throw an exception. If you legitimately have duplicate ids,
+     *     use <code>{@link #withId(String)}.{@link #first()}</code> instead.
+     *     (Note, this alternate usage is the former behavior of this selector.)
+     * </p>
+     *
+     * @param id
+     *            the id to look up
+     * @return the element with the given id
+     *
+     * @throws NoSuchElementException
+     *             if no element is found
+     */
+    public T id(String id) {
+        return withId(id).single();
+    }
+
+    /**
+     * Executes the search and returns the sole result.
+     *
+     * @return The element of the type specified in the constructor
+     * @throws NoSuchElementException
+     *             if no element is found
+     */
+    public T single() {
+        List<T> all = all();
+        if (all.size() != 1) {
+            throw new NoSuchElementException(getNoSuchElementMessage(null));
```

Comment:
I would use different messages when no elements are found and when the size is greater than one.
The message provided by `getNoSuchElementMessage(null)` is ok when no elements are found, but if we get more than one, we should not tell the developer that `No element with tag ... found`.
In that case, I would expect something like `Multiple elements with tag ... found`

Reply:
Good catch!